### PR TITLE
Evaluate code with the configured timeout

### PR DESCRIPTION
**Pull request by dialelo**
Solved #540 

@raulraja please take a look.

### CODE REVIEW

**Comment by raulraja** [comment]
LGTM but let's review the previous evaluator code to see if there were any additional security guards that we may be have missed last night when we rush to fix the threading issues in the old evaluator.

File: runtime/src/main/scala/org/scalaexercises/exercises/Evaluator.scala [hunk 1]
```diff
@@ -97,4 +99,15 @@ class Evaluator(timeout: Duration = 20.seconds) {
...
+
+  def apply[T](pre: String, code: String): EvalResult[T] = {
+    val allCode = s"""
+                     |$pre
+                     |$code
+                   """.stripMargin
+    val fut = Future({ eval(allCode) })
+    Try(
+      Await.result(fut, timeout)
+    ).getOrElse(EvalResult.Timeout())
```
Comment (maasg) [inline runtime/src/main/scala/org/scalaexercises/exercises/Evaluator.scala]: With this construction, I think we might be loosing any failure information from the execution of `eval(...)` and replacing it by a timeout. Don't know the codebase to tell if that should be the intention.

Comment (dialelo) [inline runtime/src/main/scala/org/scalaexercises/exercises/Evaluator.scala]: Good point! The `EvalResult` returned by the `Future` already carries an exception if it occurs during the execution of the code, the `Await.result` will throw an exception only for the timeouts.

File: runtime/src/test/scala/com/fortysevendeg/exercises/EvaluatorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import scala.concurrent.duration._
+import org.scalatest._
+
+class EvaluatorSpec extends FunSpec with Matchers {
+
+  describe("evaluation") {
+    it("fails with a timeout when takes longer than the configured timeout") {
+      val evaluator = new Evaluator(1 second)
+      val result: EvalResult[Int] = evaluator("", "{ while(true) {}; 123 }")
```
Comment (maasg) [inline runtime/src/test/scala/com/fortysevendeg/exercises/EvaluatorSpec.scala]: what I meant above:
```
evaluator("", "throw new Exception("bum")") should be ______?
```

Comment (dialelo) [inline runtime/src/test/scala/com/fortysevendeg/exercises/EvaluatorSpec.scala]: In that case it'd be `EvalRuntimeError` containing the thrown exception, which is of type `EvalResult[T]`. The original exception is captured and its message shown to the user in the UI.

**Comment by raulraja** [comment]
LGTM, let's address the security concerns in a different PR

**Pull request merged**